File: mlir/examples/dsp/SimpleBlocks/Output/TryDSPApps/BenchmarkTest/DSP-DSL/targetDetection.py
def main() {
  var fs = 1000;
  # var step = 1/fs; 
  # print(step);
	var input = getRangeOfVector(0, 100000000, 0.000125);
  var pi = 3.14159265359;
  var getMultiplier = 2 * pi * 10;
  # print(getMultiplier);
  var getSinDuration = gain(input, getMultiplier);
  var sig1 = sin(getSinDuration );
  var getMultiplier2 = 2 * pi * 20;
  var getSinDuration2 = gain(input, getMultiplier2);
  var sinsig2 = sin(getSinDuration2);
  var sig2 = gain(sinsig2, 0.5);
  var signal = sig1 + sig2;
  var noise = delay(signal, 5);
  var noisy_sig = signal + noise;
  
  var mu = 0.01;
  var filterSize = 20;
  var y = lmsFilterResponse(noisy_sig, signal, mu, filterSize);
  var peaks = find_peaks(y, 1, 50); 
  var final1 = getElemAtIndx(peaks , [1]); 
  var final2 = getElemAtIndx(peaks , [2]); 
  print(final1);
  print(final2);
}

File: mlir/examples/dsp/SimpleBlocks/Output/TryDSPApps/BenchmarkTest/Matlab/targetDetection.m
% Constants
INPUT_LENGTH = 100000000;
MAX_PEAKS = 100;

% Generate input range
input = (0:0.000125:(INPUT_LENGTH-1)*0.000125)';

% Generate signals
getMultiplier = 2 * pi * 10;
getSinDuration = input * getMultiplier;
sig1 = sin(getSinDuration);

getMultiplier2 = 2 * pi * 20;
getSinDuration2 = input * getMultiplier2;
sinsig2 = sin(getSinDuration2);
sig2 = 0.5 * sinsig2;

% Combine signals
signal = sig1 + sig2;

% Add delayed noise
noise = [zeros(5, 1); signal(1:end-5)];
noisy_sig = signal + noise;

% LMS Filter
mu = 0.01;
filterSize = 20;
y = lmsFilterResponse(noisy_sig, signal, mu, filterSize);

% Find peaks
[peaks, ~] = findpeaks(signal, 'MinPeakHeight', 1, 'MinPeakDistance', 50);

% Display results
fprintf('%d %d\n', peaks(2), peaks(3));


% LMS Filter Response Function
function output = lmsFilterResponse(noisy_sig, clean_sig, mu, filterSize)
    length = numel(noisy_sig);
    w = zeros(filterSize, 1);
    output = zeros(length, 1);
    
    for n = 1:length
        x = noisy_sig(max(1, n-filterSize+1):n);
        x = [zeros(filterSize - numel(x), 1); x];
        y = w' * x;
        e = clean_sig(n) - y;
        w = w + mu * e * x;
        output(n) = e;
    end
end